visualizer/src/Object.cpp

```diff
@@ -116,17 +116,28 @@ void Object::move()
     position += velocity.z * -front;
 }
 
-void Object::update()
+void Object::update(int rquat)
 {
-    update(glm::mat4(1.0f));
+    if(rquat != NULL)
+    {
+        update(glm::mat4(1.0f), rquat);
+    }
+    else
+    {
+        update(glm::mat4(1.0f));
+    }
 }
 
-void Object::update(glm::mat4 pmat)
+void Object::update(glm::mat4 pmat, int rquat)
 {
-    quat = ry * rz * rx;
+    if(rquat == NULL)
+    {
+        quat = ry * rz * rx;
+    }
 
     glm::mat4 t = glm::translate(glm::mat4(1.0f), position);
-    glm::mat4 r = glm::toMat4(quat);
+    //std::cout << this->quat << std::endl; 

```

Comment:
I did not do any quaternion mapping yet. I need help with creating values for that. Here are the new additions to the Object::update() methods to take in consideration of a quaternion vector. The quaternion vector would be set manually and then be converted into a 4x4 rotation matrix. The logic is still the same as before, but I am not sure if it is totally accurate or if it works.

Reply:
Disregard initial message, the method works. I also created random quaternion data as well.